scormplayer/scorm-impl/service/src/java/org/sakaiproject/scorm/service/impl/ScormEntityProducer.java

```diff
@@ -0,0 +1,476 @@
+        }
+
+        try {
+            ContentPackageManifest manifest = contentPackageManifestDao.load(manifestId);
+            if (manifest == null) {
+                return null;
+            }
+            ContentPackageManifest clone = SerializationUtils.clone(manifest);
+            clone.setId(null);
+
+            List<LaunchData> launchDataList = clone.getLaunchData();
+            if (launchDataList != null) {
+                launchDataList.forEach(ld -> ld.setId(null));
+            }
+
+            return contentPackageManifestDao.save(clone);
+        } catch (Exception e) {
+            log.warn("Unable to clone manifest {}: {}", manifestId, e.getMessage());
+            return null;
+        }
+    }
+
+    private void recordReferenceMapping(Map<String, String> transversalMap, String fromContext, String toContext, ContentPackage source, ContentPackage copy) {
+        if (source.getContentPackageId() != null && copy.getContentPackageId() != null) {
+            transversalMap.put(buildPackageReference(fromContext, source.getContentPackageId()),
+                    buildPackageReference(toContext, copy.getContentPackageId()));
+        }
+
+        if (source.getResourceId() != null && copy.getResourceId() != null) {
+            transversalMap.put(buildResourcePath(source.getResourceId()), buildResourcePath(copy.getResourceId()));
+            transversalMap.put(buildCollectionPath(source.getResourceId()), buildCollectionPath(copy.getResourceId()));
+        }
+    }
+
+    private void updateCollectionDisplayName(String resourceId, String title) {
+        if (title == null || title.isBlank()) {
+            return;
+        }
+
+        String path = buildCollectionPath(resourceId);
+
+        try {
+            ContentCollection collection = contentHostingService.getCollection(path);
+            ResourceProperties props = collection.getProperties();
+            String displayName = props.getProperty(ResourceProperties.PROP_DISPLAY_NAME);
+            if (title.equals(displayName)) {
+                return;
+            }
+        } catch (Exception e) {
+            log.debug("Unable to read collection properties for {}: {}", resourceId, e.getMessage());
+            return;
+        }
+
+        ContentCollectionEdit edit = null;
+        try {
+            edit = contentHostingService.editCollection(path);
+            edit.getPropertiesEdit().addProperty(ResourceProperties.PROP_DISPLAY_NAME, title);
+            contentHostingService.commitCollection(edit);
+        } catch (Exception e) {
+            if (edit != null && edit.isActiveEdit()) {
+                contentHostingService.cancelCollection(edit);
+            }
+            log.debug("Unable to update collection display name for {}: {}", resourceId, e.getMessage());
+        }
+    }
+
+    private <T> T runWithAdvisor(Callable<T> task) throws Exception {
+        SecurityAdvisor advisor = (userId, function, reference) -> SecurityAdvice.ALLOWED;
```

Comment:
there already is a stub in SecurityService for allow all permissions.

also should all permissions be allowed? (seems very wide)